oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/alarm/AlarmRecord.java

```diff
@@ -46,10 +47,11 @@
     public static final String ID1 = "id1";
     public static final String START_TIME = "start_time";
     public static final String ALARM_MESSAGE = "alarm_message";
+    public static final String RULE_NAME = "rule_name";
 
     @Override
     public String id() {
-        return getTimeBucket() + Const.ID_CONNECTOR + scope + Const.ID_CONNECTOR + id0 + Const.ID_CONNECTOR + id1;
+        return getTimeBucket() + Const.ID_CONNECTOR + scope + Const.ID_CONNECTOR + ruleName + Const.ID_CONNECTOR + id0 + Const.ID_CONNECTOR + id1;
```

Comment:
I think once you have the `ruleName`, `scope` is not necessary? RuleName matches the metrics name already, right?

Reply:
That's true, deleted.